yajsapi/market/config.ts

```diff
@@ -31,16 +31,18 @@ export class DemandConfig {
   public readonly proposalTimeout?: number;
   public readonly logger?: Logger;
   public readonly eventTarget?: EventTarget;
+  public readonly httpAgent: Agent;
 
   constructor(options?: DemandOptions) {
     const apiKey = options?.yagnaOptions?.apiKey || EnvUtils.getYagnaAppKey();
     if (!apiKey) throw new Error("Api key not defined");
     const basePath = options?.yagnaOptions?.basePath || EnvUtils.getYagnaApiUrl();
+    this.httpAgent = new Agent({ keepAlive: true });
```

Comment:
It would be best if a single agent instance would be created and passed down via constructors of these objects. Right now we produce few of them, which means we need to maintain all instances and ensure that they get closed. Consider:

```ts
constructor(
  private readonly httpAgent: Agent
  options?: DemandOptions,
)
```

It will also make it more "mockable" later in tests.

Reply:
Yes you are right. The only problem is that in the original idea we assumed that core entities such as Agreemnet, Activity, etc. can exist independently (mid-level API). With your approach we would have to create one object at the top (in Executor) and pass it down. In addition, destroy would also have to be done on top.. 
This will cause each entity and service to have an additional dependency on the constructor and maybe there would be nothing wrong with it, but I am not sure if it will be needed in future long pooling and the user (who would use services and maybe low-entities) should know what is httpAgent so for now I would leave it that way

Reply:
We should also consider making a global config object for connections to yagna to simplify mocking in tests. Each entity uses the same connection and creates it for itself every time.. So maybe with these fixes it can be improved